src/schema/autocompletes.ts

```diff
@@ -0,0 +1,59 @@
+import { AutocompleteType, Autocomplete } from '../entity';
+
+import { traceResolvers } from './trace';
+import { ILike } from 'typeorm';
+import { AuthContext, BaseContext } from '../Context';
+import { toGQLEnum } from '../common';
+import { queryReadReplica } from '../common/queryReadReplica';
+import { autocompleteSchema } from '../common/schema/autocompletes';
+import { ValidationError } from 'apollo-server-errors';
+import type z from 'zod';
+
+interface AutocompleteData {
+  result: string[];
+}
+
+export const typeDefs = /* GraphQL */ `
+  ${toGQLEnum(AutocompleteType, 'AutocompleteType')}
+
+  type AutocompleteData {
+    result: [String]!
+  }
+
+  extend type Query {
+    """
+    Get autocomplete based on type
+    """
+    autocomplete(type: AutocompleteType!, query: String!): AutocompleteData!
+      @auth
+      @cacheControl(maxAge: 3600)
+  }
+`;
+
+export const resolvers = traceResolvers<unknown, BaseContext>({
+  Query: {
+    autocomplete: async (
+      _,
+      payload: z.infer<typeof autocompleteSchema>,
+      ctx: AuthContext,
+    ): Promise<AutocompleteData> => {
+      const { data, error } = autocompleteSchema.safeParse(payload);
```

Comment:
```suggestion
      const { data } = autocompleteSchema.parse(payload);
```

you can just use `parse`, API will now convert the error thrown by it automatically to zod error, no need to throw ValidationError below